op-node/rollup/engine/payload_success.go

```diff
@@ -58,28 +55,8 @@ func (eq *EngDeriver) onPayloadSuccess(ctx context.Context, ev PayloadSuccessEve
 		eq.TryUpdateLocalSafe(ctx, ev.Ref, ev.Concluding, ev.DerivedFrom)
 	}
 	// Now if possible synchronously call FCU
-	eq.TryUpdateEngine(ctx, TryUpdateEngineEvent{
-		BuildStarted:  ev.BuildStarted,
-		InsertStarted: ev.InsertStarted,
-		Envelope:      ev.Envelope,
-	})
-}
-
-func (d *EngDeriver) TryUpdateEngine(ctx context.Context, x TryUpdateEngineEvent) {
-	// If we don't need to call FCU, keep going b/c this was a no-op. If we needed to
-	// perform a network call, then we should yield even if we did not encounter an error.
-	if err := d.ec.TryUpdateEngine(d.ctx); err != nil && !errors.Is(err, ErrNoFCUNeeded) {
-		if errors.Is(err, derive.ErrReset) {
-			d.emitter.Emit(ctx, rollup.ResetEvent{Err: err})
-		} else if errors.Is(err, derive.ErrTemporary) {
-			d.emitter.Emit(ctx, rollup.EngineTemporaryErrorEvent{Err: err})
-		} else {
-			d.emitter.Emit(ctx, rollup.CriticalErrorEvent{
-				Err: fmt.Errorf("unexpected TryUpdateEngine error type: %w", err),
-			})
-		}
-	} else if x.triggeredByPayloadSuccess() {
-		logValues := x.getBlockProcessingMetrics()
-		d.log.Info("Inserted new L2 unsafe block", logValues...)
+	err := eq.TryUpdateEngine(ctx)
```

Comment:
May I ask, previously based on this error we manually emitted events (ResetEvent, EngineTemporaryErrorEvent, CriticalErrorEvent). Now, as my understanding we do not have this fine grained error handling based on events. May I ask why, or I may be misunderstanding.

Reply:
Those emissions are still produced, just from [EngineController.OnEvent](https://github.com/ethereum-optimism/optimism/pull/17015/files?diff=split#diff-5366b09b247d2be40bf57a97e3b0e0be285517a7fcde198e667666214ca2c3a2R589) now.

- In EngineController.OnEvent handling of TryUpdateEngineEvent, errors from TryUpdateEngine are mapped exactly as before:
  - ErrReset → emit rollup.ResetEvent
  - ErrTemporary → emit rollup.EngineTemporaryErrorEvent
  - anything else → emit rollup.CriticalErrorEvent
  - If triggered by payload success, it logs the “Inserted new L2 unsafe block” message with the same metrics.

- Additionally, TryUpdateEngine itself still emits a CriticalErrorEvent only in the “unsafe head behind finalized” invariant violation, matching prior behavior.